components/gitpod-db/src/typeorm/entity/db-project-env-vars.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { PrimaryColumn, Entity, Column } from "typeorm";
+import { TypeORM } from "../typeorm";
+import { ProjectEnvVarWithValue } from "@gitpod/gitpod-protocol";
+import { Transformer } from "../transformer";
+import { encryptionService } from "../user-db-impl";
+
+@Entity()
+// on DB but not Typeorm: @Index("ind_lastModified", ["_lastModified"])   // DBSync
+export class DBProjectEnvVar implements ProjectEnvVarWithValue {
+    @PrimaryColumn(TypeORM.UUID_COLUMN_TYPE)
+    id: string;
+
+    // projectId must be part of the primary key as otherwise malicious users could overwrite
```

Comment:
The explanaition does not make sense to me: It has nothing to do with `TypeORM.save`, but only with the fact that wee need to track (and check) ownership of env vars, no? Having it as part of the primary key is a nice-to-have, 2nd line of defense and is fine with me.

That being said, my first thought was to re-use the existing table. The fact that it's called `...User...`, and it containing a field `userId` is a bit unfortunate, but could also be solved with migrations. WDYT? @jankeromnes

Reply:
> The explanaition does not make sense to me

Ah, I simply copied it over from here: 😅

https://github.com/gitpod-io/gitpod/blob/d114f037a1c34df5206bb93fc9fe48481f28bfd7/components/gitpod-db/src/typeorm/entity/db-user-env-vars.ts#L19-L24

Happy to amend or remove it. 👍

> That being said, my first thought was to re-use the existing table.
> [...]
> Update: ah, now I think I see the indent - (partly) deprecation of the UserEnvVar table. Nvm then.

Yes, (part) deprecation of UserEnvVar is indeed the intent 👍

Reply:
> Happy to amend or remove it. +1

Let's compress it to:
```
// `projectId` is part of the PK for safety reasons: This way it's impossible that a user (maliciously or by chance) sends us an env-var that has the same PK (`id`) as the env var from anther user.
```